src/psyclone/tests/parse/module_manager_test.py

```diff
@@ -343,3 +344,133 @@ def test_mod_manager_add_ignore_modules():
     # Just in case verify that other modules are not affected
     mod_info = mod_man.get_module_info("b_mod")
     assert mod_info.filename == "d1/d3/b_mod.F90"
+
+
+@pytest.mark.usefixtures("change_into_tmpdir", "clear_module_manager_instance",
+                         "mod_man_test_setup_directories")
+def test_mod_manager_add_files_and_more():
+    '''Fixture will create the following files
+
+    d1/a_mod.f90
+    d1/d3/b_mod.F90
+    d1/d3/c_mod.x90
+    d2/d_mod.X90
+    d2/d4/e_mod.F90
+
+    We will check through some further ModuleManager functions.
+    '''
+    mod_man = ModuleManager.get()
+
+    #
+    # Test add_files(...)
+    #
+    mod_man.add_files("d1/a_mod.f90")
+
+    # Add same file again, will be silently ignored
+    mod_man.add_files("d1/a_mod.f90")
+
+    #
+    # Test various other functions
+    #
+    mod_man.load_all_source_files()
```

Comment:
Please add checks that these calls have modified the internal state of `mod_man` as you expect.